popup/index.tsx

```diff
@@ -1,26 +1,301 @@
-import { useState } from "react"
+import * as ethers from "ethers"
+import { useCallback, useEffect, useState, type MouseEvent } from "react"
+
+import { BackgroundDirectMessenger } from "~packages/messenger/background/direct"
+import { WaalletContentProvider } from "~packages/provider/waallet/content/provider"
 
 import "~style.css"
 
+const provider = new ethers.BrowserProvider(
+  new WaalletContentProvider(new BackgroundDirectMessenger())
+)
+
 function IndexPopup() {
-  const [data, setData] = useState("")
+  const [accountIndex, setAccountIndex] = useState<number>(0)
+  const [accounts, setAccounts] = useState<string[]>([""])
+  const [balances, setBalances] = useState<bigint[]>([0n])
+  const [transactionHashes, setTransactionHashes] = useState<string[]>([""])
+  const [internalTransactionHashes, setInternalTransactionHashes] = useState<
+    string[]
+  >([""])
+  const [explorerUrl, setExplorerUrl] = useState<string>("")
+
+  useEffect(() => {
+    const _accountIndex = 0
+    setAccountIndex(_accountIndex)
+
+    const asyncFn = async () => {
+      const _explorerUrl = getExplorerUrl((await provider.getNetwork()).name)
+      setExplorerUrl(_explorerUrl)
+
+      const _accounts = (await provider.listAccounts()).map(
+        (account) => account.address
+      )
+      setAccounts(_accounts)
+
+      const _balances = await Promise.all(
+        _accounts.map(async (account) => {
+          return await provider.getBalance(account)
+        })
+      )
+      setBalances(_balances)
+
+      const _transactionHashes = await accountTransactions(
+        _explorerUrl,
+        _accounts[_accountIndex]
+      )
+      setTransactionHashes(_transactionHashes)
+
+      const _internalTransactionHashes = await accountInternalTransactions(
+        _explorerUrl,
+        _accounts[_accountIndex]
+      )
+      setInternalTransactionHashes(_internalTransactionHashes)
+    }
+
+    asyncFn()
+  }, [])
 
   return (
-    <div className="flex flex-col p-16">
-      <h2>
-        Welcome to your
-        <a href="https://www.plasmo.com" target="_blank">
-          {" "}
-          Plasmo
-        </a>{" "}
-        Extension!
-      </h2>
-      <input onChange={(e) => setData(e.target.value)} value={data} />
-      <a href="https://docs.plasmo.com" target="_blank">
-        View Docs {process.env.PLASMO_PUBLIC_SITE_URL}
-      </a>
-    </div>
+    <>
+      <AccountAddress
```

Comment:
I saw most of the components return empty in the else branch. Maybe we could move out the `if` check to here, make components themselves naive.